ledger-core/inc/core/database/Migrations.hpp

```diff
@@ -102,15 +102,15 @@ namespace ledger {
                   // we’re in sync with the database; perform the rollback normally
                   rollback<version, T>(sql);
 
-                  if (currentVersion >= 0) {
-                      // after rolling back this migration, we won’t have anything left, so we only
-                      // update the version for > 0
-                      if (currentVersion != 0) {
-                          auto prevVersion = version - 1;
-                          sql << "UPDATE __database_meta__ SET id = :id, version = :version", soci::use(T::COIN_ID), soci::use(prevVersion);
-                      }
+                  if (currentVersion > 0) {
+                      // if the migration number is greater than 0, we just perform the regular migration
+                      auto constexpr prevVersion = version - 1;
+                      sql << "UPDATE __database_meta__ SET version = :version WHERE id = :id", soci::use(prevVersion), soci::use(T::COIN_ID);
 
-                      Migration<version - 1, T>::backward(sql, currentVersion - 1);
+                      Migration<prevVersion, T>::backward(sql, currentVersion - 1);
+                  } else if (currentVersion == 0) {
```

Comment:
I am questioning myself about the type of `version`; why are we using `int` ? Do we allow negative version number ?
If not, I think you can change the type to any unsigned integer and change your `else if` in favor to a simple `else`.

Reply:
We use negative versions to encode non-existent data. We could use `0` instead, so I’ll make a switch.

Reply:
Since the version is now an unsigned integer type and the first `if` statement checks all the values except zero; the `else if` should become and `else` because - in the current state of the code - the `if/else` statement suggests we can have another available values to check which is not the case.